### PR TITLE
Start pipeline with data prep

### PR DESCRIPTION
**Pull request by joverlee521**
### Description of proposed changes
Adds script to prepare data for analysis, Cram tests for the script, and the start of the Snakemake pipeline that includes the data download and data prep rules. 

See commits for details. 

### Testing
The [new CI workflow](https://github.com/nextstrain/forecasts-ncov/runs/7158281061?check_suite_focus=true) which runs the new Cram tests has passed.

### CODE REVIEW

File: Snakefile [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+            data_provenance=config["data_provenances"],
+            geo_resolution=config["geo_resolutions"]
+        )
+
+rule download_public_data:
+    message: "Downloading case counts and Nextstrain clade counts from data.nextstrain.org"
+    wildcard_constraints:
+        geo_resolution = "global|usa"
+    output:
+        cases = "data/public/{geo_resolution}/cases.tsv.gz",
```
Comment (trvrb) [inline Snakefile:30]: I don't feel too strongly here, but I might have expected these to be `/data/open/global/cases.tsv.gz` and `/data/gisaid/global/cases.tsv.gz`, etc... to match how we label things in `ncov` phylogenetics workflow. I think this is more descriptive than `public` vs `private` and again matches prior expectation.

File: Snakefile [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        geo_resolution = "global|usa"
+    output:
+        cases = "data/private/{geo_resolution}/cases.tsv.gz",
+        nextstrain_clades = "data/private/{geo_resolution}/nextstrain_clades.tsv.gz"
+    params:
+        cases_url = "s3://nextstrain-data-private/files/workflows/forecasts-ncov/{geo_resolution}/cases.tsv.gz",
+        nextstrain_clades_url = "s3://nextstrain-data-private/files/workflows/forecasts-ncov/{geo_resolution}/nextstrain_clades.tsv.gz"
+    shell:
+        """
+        aws s3 cp --no-progress {params.cases_url:q} {output.cases}
```
Comment (trvrb) [inline Snakefile:53]: I sourced environment variables `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` that used for the `ncov` repo. However, this resulted in:
```
fatal error: An error occurred (403) when calling the HeadObject operation: Forbidden
```
Looking at the readme, I thought out that I needed to also add `AWS_DEFAULT_REGION`. However, even with this included, I'm still getting the same error. I'm assuming this is something going on with bucket policy for `nextstrain-data-private`. I'll follow up here on Slack as better to discuss security related issues not in public forum.

**Review by trvrb — CHANGES_REQUESTED** [review]
Really fantastic work here @joverlee521. I very much appreciate how Snakemake is set up with a config file that can be overridden and things like `global_excluded_locations` as simple text file.

I also appreciate how `prepare-data.py` is nicely verbose in reporting what parameters were used and in what the resulting locations and clades looked like.

I just have these two specific issues.

File: Snakefile (hunk 2 above)
Comment (joverlee521) [reply Snakefile:53]: Resolved permission issue in [Slack](https://bedfordlab.slack.com/archives/C036L7V615F/p1657216465039129).

File: Snakefile (hunk 1 above)
Comment (joverlee521) [reply Snakefile:30]: Right, I debated on whether to continue the GISAID/GenBank paradigm but went with `public` vs `private` to match the underlying public URL vs private S3 bucket. I don't feel too strongly either way, so I can update to `open` vs `gisaid`.

Comment (joverlee521) [reply Snakefile]: Updated in 3f44c62

Comment (trvrb) [reply Snakefile]: Ah! I see your logic. I guess I was imagining eventual URLs of things like nextstrain.org/forecasts/ncov/gisaid/global and wanting to mirror what users see. Thanks so much for making the change.

File: Snakefile (hunk 2 above)
Comment (trvrb) [reply Snakefile:53]: Thank you for taking care of this. 🙏

**Review by trvrb — APPROVED** [review]
Thanks so much for attending to changes requested @joverlee521.

**Pull request merged**